subsys/spm/spm.c

```diff
@@ -135,8 +135,21 @@ static void spm_config_nsc_flash(void)
 }
 #endif /* CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS */
 
+extern int spm_secure_services_init(void);
+
 static void spm_config_flash(void)
 {
+	if (((u32_t)_image_rom_start + (u32_t)_flash_used - 1)
+		>= ((LAST_SECURE_REGION_INDEX + 1)
+			* FLASH_SECURE_ATTRIBUTION_REGION_SIZE)) {
+		PRINT("End of SPM: 0x%x\n",
+			((u32_t)_image_rom_start + (u32_t)_flash_used) - 1);
+		PRINT("Start of app: 0x%x\n", FIRST_NONSECURE_ADDRESS);
+		PRINT("SPM and app are sharing an SPU region. Cannot partition "
+			"flash correctly into secure and non-secure, spinning!\n");
+		while (1)
+			;
```

Comment:
Could we enforce this through PM or through a LD / CPP error?

(To save space)

Reply:
CPP: Not possible since it depends on a linker symbol
LD: Hard since the linker symbol is defined at the very end of the linker script.

Any tips here would be welcome!

Reply:
@hakonfam : Is it possible to enforce the above through PM?

Reply:
I tried adding an assert now through `zephyr_linker_sources()` and it works :D, even though the assert statement comes before the symbol is defined.

Reply:
It would be possible for PM to check this, whether its a good idea or not is not something I have considered.

Reply:
Very nice.